### PR TITLE
chore(angular): upgrade to angular 12

### PR DESCRIPTION
**Pull request by siddharthcr**
Closes #

**Summary**

Upgrading `@ai-apps/angular` to angular 12

**Change List (commits, features, bugs, etc)**

- upgrade to angular 12
- enabled 'partial' compilation mode with ivy
- refactored code to remove circular dependency
- upgraded storybook for to 6.5.9 and webpack 5
- refactored stories to use new story format

**Acceptance Test (how to verify the PR)**

- tests here

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/angular/.storybook/main.js [hunk 1]
```diff
@@ -1,29 +1,73 @@
...
+function isStylingRule(rule) {
+  const { test } = rule;
+  if (!test) {
+      return false;
+  }
+  if (!(test instanceof RegExp)) {
+      return false;
+  }
+  return test.test('.css') || test.test('.scss') || test.test('.sass');
+}
```
Comment (zvonimirfras) [inline packages/angular/.storybook/main.js:73]:
```suggestion
}

```

File: packages/angular/.storybook/main.js [hunk 2]
```diff
@@ -1,29 +1,73 @@
...
 };
+
+function isStylingRule(rule) {
+  const { test } = rule;
+  if (!test) {
+      return false;
+  }
+  if (!(test instanceof RegExp)) {
+      return false;
+  }
```
Comment (zvonimirfras) [inline packages/angular/.storybook/main.js:71]:
```suggestion
  if (!test || !(test instanceof RegExp)) {
      return false;
  }
```

File: packages/angular/package.json [hunk 3]
```diff
@@ -33,37 +34,34 @@
...
   "peerDependencies": {
-    "@angular/common": "^9.0.0 || ^10.0.0 || ^11.0.0",
-    "@angular/core": "^9.0.0 || ^10.0.0 || ^11.0.0",
-    "@angular/forms": "^9.0.0 || ^10.0.0 || ^11.0.0",
+
     "carbon-components": "^10.0.0",
     "carbon-components-angular": "^4.0.0",
-    "rxjs": "^6.0.0",
-    "zone.js": "^0.8.26 || ^0.9.0 || ^0.10.0"
+    "rxjs": "^6.0.0"
```
Comment (zvonimirfras) [inline packages/angular/package.json:40]: `peerDependencies` should stay peer dependencies. Angular dependencies should be extended to include 12 (and maintain backwards compatibility).

rxjs should probably stay here and be removed from regular dependencies.

File: packages/angular/package.json [hunk 4]
```diff
@@ -33,37 +34,34 @@
...
+    "@angular/animations": "12.2.16",
+    "@angular/cli": "12.2.17",
+    "@angular/common": "12.2.16",
+    "@angular/compiler": "12.2.16",
+    "@angular/compiler-cli": "12.2.16",
+    "@angular/core": "12.2.16",
+    "@angular/forms": "12.2.16",
+    "@angular/platform-browser": "12.2.16",
+    "@angular/platform-browser-dynamic": "12.2.16",
+    "@angular/router": "12.2.16",
```
Comment (zvonimirfras) [inline packages/angular/package.json:54]: Dev deps (all, not just highlighted) should stay at lower versions to ensure backwards compatibility when adding new things. Integration tests cover everything else.
_It's ok to change them temporarily for testing locally._

File: packages/angular/package.json [hunk 5]
```diff
@@ -33,37 +34,34 @@
...
-    "@storybook/theming": "^6.2.3",
+    "@storybook/addon-actions": "6.5.9",
+    "@storybook/addon-essentials": "6.5.9",
+    "@storybook/addon-links": "6.5.9",
+    "@storybook/angular": "6.5.9",
+    "@storybook/builder-webpack5": "6.5.9",
+    "@storybook/manager-webpack5": "6.5.9",
+    "@storybook/source-loader": "6.5.9",
+    "@storybook/addon-storysource": "6.5.9",
+    "@storybook/theming": "6.5.9",
```
Comment (zvonimirfras) [inline packages/angular/package.json:64]: It's good that these are locked versions. 👍 
@Akshat55, should they be locked at 6.2.3 instead?

File: packages/angular/src/date-time-picker/date-time-picker.component.ts [hunk 6]
```diff
@@ -13,61 +13,7 @@ import { format, isThisMinute, subHours, subMinutes } from 'date-fns';
 import { getRangeFromRelative } from './date-time-relative.component';
 import * as languages from 'flatpickr/dist/l10n/index';
 import { I18n } from 'carbon-components-angular/i18n';
-
```
Comment (zvonimirfras) [inline packages/angular/src/date-time-picker/date-time-picker.component.ts:16]: Moving the exports changes how this is used (API) and is considered a breaking change. Can't do that in a non-major version.

Re-exporting from here should be ok if you wanna keep it out, as a preparation for the major.

File: packages/angular/src/list/ai-list.component.ts [hunk 7]
```diff
@@ -2,12 +2,7 @@ import { Component, EventEmitter, Input, OnInit, Output, TemplateRef } from '@an
 import { AIListItem } from './list-item/ai-list-item.class';
 import { IconService } from 'carbon-components-angular';
 import { Bee32 } from '@carbon/icons';
-
```
Comment (zvonimirfras) [inline packages/angular/src/list/ai-list.component.ts:5]: Same as date time exports.

File: packages/angular/src/package.json [hunk 8]
```diff
@@ -16,10 +16,10 @@
   "license": "Apache-2.0",
   "author": "IBM",
   "peerDependencies": {
-    "@angular/common": "^9.0.0 || ^10.0.0 || ^11.0.0",
-    "@angular/core": "^9.0.0 || ^10.0.0 || ^11.0.0",
+    "@angular/common": "^12.0.0 || ^13.0.0 || ^14.0.0",
+    "@angular/core": "^12.0.0 || ^13.0.0 || ^14.0.0",
```
Comment (zvonimirfras) [inline packages/angular/src/package.json:20]: Looks like a breaking change

File: packages/angular/src/package.json [hunk 9]
```diff
@@ -16,10 +16,10 @@
...
     "carbon-components": "^10.0.0",
-    "carbon-components-angular": "^4.0.0"
+    "carbon-components-angular": "^5.0.1"
```
Comment (zvonimirfras) [inline packages/angular/src/package.json:22]: Looks like a breaking change

**Review by zvonimirfras — CHANGES_REQUESTED** [review]
`chore` doesn't seem adequate. `feat` seem better once the breaking stuff is out.

File: packages/angular/src/flyout-menu/flyout-menu.stories.ts [hunk 10]
```diff
@@ -9,15 +8,48 @@ import {
 
 import { FlyoutMenuModule } from './flyout-menu.module';
 
-storiesOf('Components/Filter menu', module)
-  .addDecorator(
+export default {
+  title: 'Components/Filter menu',
+
```
Comment (Akshat55) [inline packages/angular/src/flyout-menu/flyout-menu.stories.ts:13]:
```suggestion
```

File: packages/angular/src/icon-content-switcher/icon-content-switcher.stories.ts [hunk 11]
```diff
@@ -19,25 +17,41 @@ class AppDemoIcons {
   }
 }
 
-storiesOf('Components/Icon content switcher', module)
-  .addDecorator(
+export default {
+  title: 'Components/Icon content switcher',
+
```
Comment (Akshat55) [inline packages/angular/src/icon-content-switcher/icon-content-switcher.stories.ts:22]:
```suggestion
```

File: packages/angular/src/tabs/tabs.stories.ts [hunk 12]
```diff
@@ -26,8 +24,10 @@ class StoryIconModule {
   }
 }
 
-storiesOf('Components/Tabs', module)
-  .addDecorator(
+export default {
+  title: 'Components/Tabs',
+
```
Comment (Akshat55) [inline packages/angular/src/tabs/tabs.stories.ts:29]:
```suggestion
```

File: packages/angular/src/toolkit/page-header/page-header.stories.ts [hunk 13]
```diff
@@ -13,47 +12,67 @@ const createBreadcrumbItems = (count: number, content = 'Breadcrumb'): Breadcrum
       href: '#' + (i + 1),
     }));
 
-storiesOf('Sterling Toolkit/Page header', module)
-  .addDecorator(
+export default {
+  title: 'Sterling Toolkit/Page header',
+
```
Comment (Akshat55) [inline packages/angular/src/toolkit/page-header/page-header.stories.ts:17]:
```suggestion
```

File: packages/angular/src/toolkit/sortable-list/sortable-list.stories.ts [hunk 14]
```diff
@@ -1,41 +1,43 @@
...
-
-import { ListItem } from 'carbon-components-angular';
+import { moduleMetadata } from '@storybook/angular';
 import { SortableListModule } from './sortable-list.module';
 
-storiesOf('Sterling Toolkit/Sortable list', module)
-  .addDecorator(
+export default {
+  title: 'Sterling Toolkit/Sortable list',
+
```
Comment (Akshat55) [inline packages/angular/src/toolkit/sortable-list/sortable-list.stories.ts:6]:
```suggestion
```

File: packages/angular/src/toolkit/table-settings/table-settings.stories.ts [hunk 15]
```diff
@@ -21,138 +20,175 @@ class StoryCustomComponent {
   @Input() hello;
 }
 
-storiesOf('Sterling Toolkit/Table settings modal', module)
-  .addDecorator(
+export default {
+  title: 'Sterling Toolkit/Table settings modal',
+
```
Comment (Akshat55) [inline packages/angular/src/toolkit/table-settings/table-settings.stories.ts:25]:
```suggestion
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Comment by davidicus** [comment]
@Akshat55 @zvonimirfras any updates here. Trying to close out any stale PRs

**Comment by zvonimirfras** [comment]
I think we're not merging this at this point.
@Akshat55 double check and close, pls 🙂

**Comment by Akshat55** [comment]
@davidicus, we aren't merging this PR now as it contains breaking changes. Will be useful in future when carbon-addons-iot-react is updated to Carbon 11 though.

**Comment by davidicus** [comment]
Closing for now. Feel free to re-open when ready.

**Pull request closed without merging**